main.go

```diff
@@ -231,32 +213,40 @@ func createMilestoneData(advance int, timeInterval string) map[string]string {
 }
 
 func createMilestones(baseURL string, token string, projectID string, milestones map[string]string) error {
-	strURL := []string{baseURL, projectID, "/milestones"}
-	URL := strings.Join(strURL, "")
 	client := &http.Client{}
-
+	strURL := []string{baseURL, "/projects/", projectID, "/milestones"}
+	URL := strings.Join(strURL, "")
+	params := url.Values{}
 	for k, v := range milestones {
-		urlV := url.Values{}
-		urlV.Set("title", k)
-		urlV.Set("due_date", v)
-		mbyte := bytes.NewReader([]byte(urlV.Encode()))
-		req, err := http.NewRequest("POST", URL, mbyte)
+		params.Set("title", k)
+		params.Set("dueDate", v)
+		req, err := http.NewRequest("POST", URL, bytes.NewReader([]byte(params.Encode())))
 		if err != nil {
 			return err
 		}
-
 		req.Header.Add("PRIVATE-TOKEN", token)
 		resp, err := client.Do(req)
 		if err != nil {
 			return err
 		}
-
 		defer resp.Body.Close()
 	}
 
 	return nil
 }
 
+func createMilestoneMap(milestoneAPI []milestoneAPI) (map[string]milestone) {
+	var m milestone
+	milestones := map[string]milestone{}
+	for _, v := range milestoneAPI {
+		m.DueDate = v.DueDate
+		m.ID = strconv.Itoa(v.ID)
+		m.Title = v.Title
+	}
+
+	return milestones
```

Comment:
Does this code actually work? Did you print milestones to check?
In my view it does only return an empty milestones as you are never writing the milestone object to milestones.